src/Providers/ProvidersPage.tsx

```diff
@@ -0,0 +1,178 @@
+    filter: {
+      type: 'freetext',
+      toPlaceholderLabel: (t) => t('Filter by name'),
+    },
+    sortable: true,
+  },
+  {
+    id: NAMESPACE,
+    toLabel: (t) => t('Namespace'),
+    isVisible: true,
+    isIdentity: true,
+    filter: {
+      type: 'freetext',
+      toPlaceholderLabel: (t) => t('Filter by namespace'),
+    },
+    sortable: true,
+  },
+  {
+    id: READY,
+    toLabel: (t) => t('Ready'),
+    isVisible: true,
+    filter: {
+      type: 'enum',
+      primary: true,
+      toPlaceholderLabel: (t) => t('Ready'),
+      values: fromI18nEnum(CONDITIONS),
+    },
+    sortable: true,
+  },
+  {
+    id: URL,
+    toLabel: (t) => t('Endpoint'),
+    isVisible: true,
+    filter: {
+      type: 'freetext',
+      toPlaceholderLabel: (t) => t('Filter by endpoint'),
+    },
+    sortable: true,
+  },
+  {
+    id: TYPE,
+    toLabel: (t) => t('Type'),
+    isVisible: true,
+    filter: {
+      type: 'enum',
+      primary: true,
+      toPlaceholderLabel: (t) => t('Type'),
+      values: fromI18nEnum(PROVIDERS),
+    },
+    sortable: true,
+  },
+  {
+    id: VM_COUNT,
+    toLabel: (t) => t('VMs'),
+    isVisible: true,
+    sortable: true,
+  },
+  {
+    id: NETWORK_COUNT,
+    toLabel: (t) => t('Networks'),
+    isVisible: true,
+    sortable: true,
+  },
+  {
+    id: CLUSTER_COUNT,
+    toLabel: (t) => t('Clusters'),
+    isVisible: false,
+    sortable: true,
+  },
+  {
+    id: HOST_COUNT,
+    toLabel: (t) => t('Hosts'),
+    isVisible: true,
+    sortable: true,
+  },
+  {
+    id: STORAGE_COUNT,
+    toLabel: (t) => t('Storage'),
+    isVisible: false,
+    sortable: true,
+  },
+  {
+    id: ACTIONS,
+    toLabel: () => '',
+    isVisible: true,
+    sortable: false,
+  },
+];
+
+export const ProvidersPage = ({ namespace, kind }: ResourceConsolePageProps) => {
+  const { t } = useTranslation();
+
+  const dataSource = useProvidersWithInventory({
+    kind,
+    namespace,
+  });
+
+  // data hook triggers frequent re-renders although data remains the same:
+  // both the content content and object reference
+  return <PageMemo dataSource={dataSource} namespace={namespace} title={t('Providers')} />;
+};
+
+const Page = ({
+  dataSource,
+  namespace,
+  title,
+}: {
+  dataSource: [MergedProvider[], boolean, boolean];
+  namespace: string;
+  title: string;
+}) => (
+  <StandardPage<MergedProvider>
+    addButton={<AddProviderButton />}
+    dataSource={dataSource}
+    RowMapper={ProviderRow}
+    fieldsMetadata={fieldsMetadata}
+    namespace={namespace}
+    title={title}
+  />
+);
+
+const PageMemo = React.memo(Page);
+
+const AddProviderButton = () => {
+  const { t } = useTranslation();
+  const launchModal = useModal();
+
+  return (
+    <Button variant="primary" onClick={() => launchModal(withQueryClient(AddProviderModal), {})}>
+      {t('Add Provider')}
+    </Button>
+  );
+};
+
+const AddProviderModal = ({ closeModal }: { closeModal: () => void }) => {
```

Comment:
I really don't like the add provider being a modal.  It should be another page just like Add Pod etc.  The lack of screen space etc is just annoying.
/rant

Reply:
Right. The plan is to move it to a page under the details.